### ISSUE TITLE
Group norm for 3D images

### ISSUE THREAD

**Issue by soheilesm** [taki0112/Group_Normalization-Tensorflow#3]
I have a grayscale 3D image where I have let's say N samples with dims of H, W, D and C=1, what would be the proper way of normalizing it using your group norm routine?

**Comment by soheilesm** [comment]
This is the way I did it.

```
def group_norm(x, G=32, eps=1e-5, scope='group_norm') :
    with tf.variable_scope(scope) :
        N, D, H, W, C = x.get_shape().as_list()      
        G = min(G, C)

        x = tf.reshape(x, [-1, D, H, W, G, C // G])
        mean, var = tf.nn.moments(x, [1, 2, 3, 5], keep_dims=True)
        x = (x - mean) / tf.sqrt(var + eps)
        # with tf.variable_scope("group_norm", reuse=tf.AUTO_REUSE):
        gamma = tf.get_variable('gamma', [1, 1, 1, 1, C], initializer=tf.constant_initializer(1.0))
        beta = tf.get_variable('beta', [1, 1, 1, 1, C], initializer=tf.constant_initializer(0.0))
        x = tf.reshape(x, [-1, D, H, W, C]) * gamma + beta

    return x

```

But if I use group norm multiple times I get dimension error for beta and gamma since they will have a different value of C along different layers. I still cannot solve the issue, using setting reuse to false, or none or doing local_variable_initializer, I could find a stupid way around it but creating multiple functions of group norm each with a different name for betta and gamma.

Can you suggest a smarter way? did you get into this issue of already existing variables gamma and beta and not sharing them?
